ui_framework/src/components/flex/_flex_grid.scss

```diff
@@ -0,0 +1,60 @@
+.kuiFlexGrid {
```

Comment:
I know you hate `calc`, but how about we try this?

```scss
.kuiFlexGrid {
  display: flex;
  align-items: stretch;
  flex-wrap: wrap;
  margin-bottom: 0;

  .kuiFlexItem {
    flex-grow: 0;
  }
}

$gutterTypes: (
  gutterSmall: $kuiSizeS / 2,
  gutterMedium: $kuiSize / 2,
  gutterLarge: $kuiSizeL / 2,
  gutterExtraLarge: $kuiSizeXL / 2,
);

$fractions: (
  flexBasisFourths: 25%,
  flexBasisThirds: 33.3%,
  flexBasisHalves: 50%,
);

@each $gutterName, $gutterSize in $gutterTypes {
  @each $fraction, $percentage in $fractions {
    $halfGutterSize: $gutterSize * 0.5;

    .kuiFlexGrid--#{$gutterName} {
      margin: -$halfGutterSize;

      &.kuiFlexGrid--#{$fraction} {
        .kuiFlexItem {
          margin: $halfGutterSize;
          flex-basis: calc(#{$percentage} - #{$gutterSize});
        }
      }
    }
  }
}
```

This will apply my trick from http://www.flexboxpatterns.com/gallery and get rid of the extra margin on the top row of FlexItems.